pkg/plugin/sdk/v1alpha2/plugin/interface.go

```diff
@@ -62,3 +73,42 @@ func (*Base) PostTestHook(bool, ioutils.PromptRunner) error {
 func (*Base) PostRunHook(bool, ioutils.PromptRunner) error {
 	return nil
 }
+
+type CustomCommandFn (func(ctx context.Context, args []string, bzl bazel.Bazel) error)
+
+type Command struct {
+	Use       string
+	ShortDesc string
+	LongDesc  string
+	Run       CustomCommandFn
+}
+
+type CommandManager interface {
+	Save(commands []*Command) error
+	Execute(command string, ctx context.Context, args []string) error // Not sure about this signature, haven't read the whole code yet.
+}
+
+type PluginCommandManager struct {
+	commands map[string]CustomCommandFn
+}
+
+func (cm *PluginCommandManager) Save(commands []*Command) error {
+	for _, cmd := range commands {
+		if _, exists := cm.commands[cmd.Use]; exists {
+			return fmt.Errorf("command '%s' is declared more than once by plugin", cmd.Use)
```

Comment:
if the command is `build` will it fail here?

Reply:
I don't think so. @JesseTatasciore we need some guard rails on the cobra side when constructing the commands.

Reply:
Check added. Sample error: `Error: failed to register custom commands: plugin implements a command with a protected name: test`